### PR TITLE
Additions for 616

### LINKED ISSUE #652: Fixes for Asset transaction Payment Pallet
ToDos
- [x] Create a Mock environment
- [x] Adds test cases for pallet
- [x] Create a new crate polkadex_extrinsic and move all the signing logic there

### PR DESCRIPTION
**Pull request by nakul1010**
closes #652.

### CODE REVIEW

File: client/Cargo.toml [hunk 1]
```diff
@@ -10,3 +10,4 @@ sc-executor = { workspace = true }
 node-polkadex-runtime = { path = "../runtime", version = "4.0.0" }
 frame-benchmarking = { workspace = true }
 thea-primitives = {path = "../thea-primitives"}
+polkadex-extrinsic = {path = "../polkadex-extrinsic"}
```
Comment (serhii-temchenko) [inline client/Cargo.toml]: Please add spaces before `}` and after `{`.

Comment (serhii-temchenko) [inline client/Cargo.toml]: Please add an empty line at the end of file

File: pallets/assets-transaction-payment/src/lib.rs [hunk 2]
```diff
@@ -189,7 +197,7 @@ where
 		if fee.is_zero() {
 			Ok((fee, InitialPayment::Nothing))
 		} else if self.asset_id != Zero::zero() {
-			// If the asset id is zero, then we treat that case as payment in PDEX,
+			//If the asset id is zero, then we treat that case as payment in PDEX,
```
Comment (serhii-temchenko) [inline pallets/assets-transaction-payment/src/lib.rs]: I assume this comment moved into the `else` case. Please remove it from here.

File: polkadex-extrinsic/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    "parity-util-mem/std",
+    "sp-core/std",
+    "polkadex-primitives/std",
+    "sp-runtime/std",
+    "parity-scale-codec/std",
+    "frame-support/std",
+    "scale-info/std",
+    "sp-io/std",
+    "serde/std",
+]
```
Comment (serhii-temchenko) [inline polkadex-extrinsic/Cargo.toml]: Please add an empty line at the end of file

File: pallets/assets-transaction-payment/Cargo.toml [hunk 4]
```diff
@@ -38,12 +44,7 @@ std = [
...
-    "frame-benchmarking/runtime-benchmarks",
-    "frame-support/runtime-benchmarks",
-    "frame-system/runtime-benchmarks",
-    "sp-runtime/runtime-benchmarks",
-]
-try-runtime = ["frame-support/try-runtime"]
+	"pallet-transaction-payment/std",
+	"pallet-timestamp/std",
+	"polkadex-extrinsic/std",
+]
```
Comment (serhii-temchenko) [inline pallets/assets-transaction-payment/Cargo.toml]: Please add empty line at the end of file

File: pallets/assets-transaction-payment/src/test.rs [hunk 5]
```diff
@@ -0,0 +1,524 @@
...
+	});
+}
+
+#[test]
+fn block_token_for_fees_when_token_not_allowlisted() {
+	new_test_ext().execute_with(|| {
+		let _asset = 1_u128;
+		//ToDo: Remove comment after fix
+		// assert_noop!(AssetsTransactionPayment::block_token_for_fees(Origin::root(),
+		// Error::<Test>::TokenNotAllowlisted));
```
Comment (serhii-temchenko) [inline pallets/assets-transaction-payment/src/test.rs]: Is it valid logic to do? I mean - isn't remove operations supposed to be idempotent? it's means - result should not throw error even if it's not found or not "allowlisted", I think logic in this extrinsic implemented in this way no?   So to test this extrinsic - there should be asserted that remove of not "allowlisted" asset will return ok result, remove of "allowlisted" asset will remove it from the collection and of course that  called by the root.

File: pallets/assets-transaction-payment/src/test.rs [hunk 6]
```diff
@@ -0,0 +1,524 @@
...
+	});
+}
+
+#[test]
+fn block_token_for_fees() {
+	new_test_ext().execute_with(|| {
+		let asset = 1_u128;
+		assert_ok!(AssetsTransactionPayment::allow_list_token_for_fees(Origin::root(), asset));
```
Comment (serhii-temchenko) [inline pallets/assets-transaction-payment/src/test.rs]: After this assertion still should be asserted that it was added into the collection e.g.:
```rust
assert_eq!(<AllowedAssets<Test>>::get(), vec![asset]);
```

File: pallets/assets-transaction-payment/src/test.rs [hunk 7]
```diff
@@ -0,0 +1,524 @@
...
+		assert_ok!(AssetsTransactionPayment::block_token_for_fees(Origin::root(), asset));
+		let vec: Vec<u128> = Vec::new();
+		assert_eq!(<AllowedAssets<Test>>::get(), vec);
+	});
+}
+pub fn info_from_weight(w: Weight) -> DispatchInfo {
```
Comment (serhii-temchenko) [inline pallets/assets-transaction-payment/src/test.rs:64]: Empty string before this line

**Review by serhii-temchenko — CHANGES_REQUESTED** [review]

**Comment by 35359595** [comment]
can we have more specific name for this crate? extrinsic should preserve one meaning from substrate to avoid any confusion.

File: pallets/assets-transaction-payment/src/test.rs (hunk 5 above)
Comment (nakul1010) [reply pallets/assets-transaction-payment/src/test.rs]: yes added both the test cases

File: pallets/assets-transaction-payment/src/test.rs (hunk 6 above)
Comment (nakul1010) [reply pallets/assets-transaction-payment/src/test.rs]: added in latest test case `block_token_for_fees_when_token_allowlisted`

File: pallets/assets-transaction-payment/src/test.rs (hunk 7 above)
Comment (nakul1010) [reply pallets/assets-transaction-payment/src/test.rs:64]: Done

**Comment by nakul1010** [comment]
@serhii-temchenko Thanks for including the CI commit code fix from the develop branch into this branch.

**Comment by nakul1010** [comment]
> can we have more specific name for this crate? extrinsic should preserve one meaning from substrate to avoid any confusion.

I assume u are mentioning the `polkadex-extrinsic` crate, can I change the name to `wallet-connect`?

**Comment by 35359595** [comment]
> > can we have more specific name for this crate? extrinsic should preserve one meaning from substrate to avoid any confusion.
> 
> I assume u are mentioning the `polkadex-extrinsic` crate, can I change the name to `wallet-connect`? 

it also sounds like action. is wallet-connector describing purpose of this crate?

**Comment by nakul1010** [comment]
> > > can we have more specific name for this crate? extrinsic should preserve one meaning from substrate to avoid any confusion.
> > 
> > 
> > I assume u are mentioning the `polkadex-extrinsic` crate, can I change the name to `wallet-connect`?
> 
> it also sounds like action. is wallet-connector describing purpose of this crate?

yes, the crate basically has the ability to send transactions using any generic wallet that is not supported by substrate. So i will go with `wallet-connector` @Gauthamastro are u fine with this.

**Review by serhii-temchenko — APPROVED** [review]

**Pull request merged**